tests/reader/reader_test.go

```diff
@@ -10,70 +12,165 @@ import (
+	t.Run("reads single file", func(t *testing.T) {
+		file1, err := tests.CreateTempFile(t, tests.SampleText32)
+		if err != nil {
+			t.Fatalf("Failed to create temporary file: %s", err)
+		}
 
-	result, err := contentReader.ReadAll()
-	assert.NoError(t, err)
-	expectedResult := content1 + "\n" + content2 + "\n"
-	assert.Equal(t, expectedResult, result)
-}
+		config := &options.Config{
+			FilePaths: []string{file1.Name()},
+		}
 
-func TestReadFile(t *testing.T) {
-	content := "file content"
-	file, err := tests.CreateTempFile(t, content)
-	if err != nil {
-		t.Fatalf("Failed to create temporary file: %s", err)
-	}
+		contentReader := reader.ContentReader{Config: config}
 
-	config := &options.Config{}
-	contentReader := reader.ContentReader{Config: config}
+		result, err := contentReader.ReadAll()
+		assert.NoError(t, err)
+		expectedResult := tests.SampleText32 + "\n"
+		assert.Equal(t, expectedResult, result)
+	})
+
+	t.Run("reads from argument", func(t *testing.T) {
+		config := &options.Config{
+			Text: tests.SampleText32,
+		}
+
+		contentReader := reader.ContentReader{Config: config}
 
-	result, err := contentReader.ReadFile(file.Name())
-	assert.NoError(t, err)
-	assert.Equal(t, content, result)
+		result, err := contentReader.ReadAll()
+		assert.NoError(t, err)
+		expectedResult := tests.SampleText32 + "\n"
+		assert.Equal(t, expectedResult, result)
+	})
+
+	t.Run("reads multiple files", func(t *testing.T) {
+		fileContents := []string{"Japan", "Australia", "Germany"}
+		expectedResults := strings.Join(fileContents, "\n") + "\n"
+		var filePaths []string
+
+		for index := 0; index < 3; index++ {
+			file, err := tests.CreateTempFile(t, fileContents[index])
+			if err != nil {
+				t.Fatalf("Failed to create temporary file: %s", err)
+			}
+
+			filePaths = append(filePaths, file.Name())
+		}
+
+		config := &options.Config{FilePaths: filePaths}
+
+		contentReader := reader.ContentReader{Config: config}
+
+		result, err := contentReader.ReadAll()
+
+		assert.NoError(t, err)
+		assert.Equal(t, expectedResults, result)
+	})
 }
 
+//nolint:funlen // It cannot be shorter
 func TestReadable(t *testing.T) {
-	file, err := tests.CreateTempFile(t, "content")
-	if err != nil {
-		t.Fatalf("Failed to create temporary file: %s", err)
-	}
-
 	config := &options.Config{}
 	contentReader := reader.ContentReader{Config: config}
 
-	err = contentReader.Readable(file.Name())
-	assert.NoError(t, err)
+	t.Run("file exists and is readable", func(t *testing.T) {
+		file, err := tests.CreateTempFile(t, "content")
+		if err != nil {
+			t.Fatalf("Failed to create temporary file: %s", err)
+		}
 
-	err = contentReader.Readable("nonexistentfile")
-	assert.Error(t, err)
+		err = contentReader.Readable(file.Name())
+		assert.NoError(t, err)
+	})
+
+	t.Run("file does not exist", func(t *testing.T) {
+		err := contentReader.Readable("nonexistentfile")
+		assert.Error(t, err)
+		assert.Equal(t, "file does not exist or can't be accessed", err.Error())
```

Comment:
(Here and everywhere)

This is a bad way.

You should return a sentinel error and use


```suggestion
		assert.ErrorIs(t, err, ErrFileNotFound)
```

Reply:
I've used [assert.EqualError()](https://github.com/stretchr/testify/blob/v1.9.0/assert/assertions.go#L1573), bc I didn't find a simple way to use the [assert.ErrorIs()](https://github.com/stretchr/testify/blob/v1.9.0/assert/assertions.go#L2031), it required two error interfaces to be passed e.g.: `assert.ErrorIs(t, err, targerErr, errMsg)`, in the other hand the `assert.EqualError(t, theError, theErrorMsg)` was more straightforward to implement it.